.changeset/wise-jeans-chew.md

```diff
@@ -0,0 +1,41 @@
+---
+"@effect/ai-openai": patch
+"@effect/ai": patch
+---
+
+Support creation of embeddings from the AI integration packages.
+
+For example, the following program will create an OpenAI `Embeddings` service
+that will aggregate all embedding requests received within a `500` millisecond
+window into a single batch.
+
+```ts
+import { Embeddings } from "@effect/ai"
+import { OpenAiClient, OpenAiEmbeddings } from "@effect/ai-openai"
+import { NodeHttpClient } from "@effect/platform-node"
+import { Config, Effect, Layer } from "effect"
+
+// Create the OpenAI client
+const OpenAi = OpenAiClient.layerConfig({
+  apiKey: Config.redacted("OPENAI_API_KEY")
+}).pipe(Layer.provide(NodeHttpClient.layerUndici))
+
+// Create an embeddings service for the `text-embedding-3-large` model
+const TextEmbeddingsLarge = OpenAiEmbeddings.layerDataLoader({
```

Comment:
will it batch if I use
```ts
(documents: Array<string>) => Effect.forEach(documents, embedding.embed, { concurrency: "unbounded" })
```
and `OpenAiEmbeddings.layer`?

Reply:
You would need to add batching true to the forEach options as under the hood it's using RequestResolver.batched. 

Unless you use the data loader variant, in which case it will always batch as much as it can during the specified window (up to the max batch size).